File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        maj1 = c1 = 0
        for i in nums:
            if maj1 == i:
                c1 += 1
            elif c1 == 0 :
                maj1 = i
                c1 += 1
            else: 
                c1 -= 1
        return maj1
        # nums.sort()
        # return nums[len(nums)//2]

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int maj = 0 , c = 0;
        for (auto i : nums){
            if (maj == i) c++;
            else if ( c == 0) {
                maj = i;
                c = 1;
            }
            else c--;
        }
        return maj;
    }
};